app/permissions/PermissionsActionCheck.scala

```diff
@@ -27,6 +27,56 @@ trait PermissionActionFilter extends ActionFilter[UserRequest] with Logging {
     }
   }
 }
+trait ModifyCollectionsPermissionsCheck extends Logging { self: BaseFaciaController =>
+
+  def acl: Acl
+  def configAgent: ConfigAgent
+
+  private def testAccess(email: String, priorities: Set[PermissionsPriority], isLaunch: Boolean) = {
+
+    if (priorities.isEmpty) AccessGranted
+    else if (isLaunch)
+      acl.testUserGroupsAndCollections(
+        Permissions.LaunchEditorialFrontsAlert, Permissions.LaunchCommercialFrontsAlert,
+        Permissions.FrontsAccess, "facia-tool-allow-access-for-all")(email, priorities)
+    else
+      acl.testUserGroupsAndCollections(
+        Permissions.EditEditorialFrontsAlert, Permissions.LaunchCommercialFrontsAlert,
+        Permissions.FrontsAccess, "facia-tool-allow-access-for-all")(email, priorities)
+  }
+
+  def withModifyGroupPermissionForCollections[A](priorities: Set[PermissionsPriority], secondaryPriorities: Set[PermissionsPriority],
+                                                 isLaunch: Boolean = false)(block: => Future[Result])
+                                                (implicit request: UserRequest[A],
+                                                 executionContext: ExecutionContext): Future[Result] = {
+
+    (testAccess(request.user.email, priorities, isLaunch), testAccess(request.user.email, secondaryPriorities, isLaunch)) match {
+      case (AccessGranted, AccessGranted) => block
+      case _ => {
+        logger.info(s"User with e-mail ${request.user.email} not authorized to modify collection")
+        Future.successful(Results.Unauthorized)
+      }
+    }
+  }
+}
+
+class LaunchCommercialFrontsPermissionCheck(val acl: Acl)(implicit ec: ExecutionContext) extends PermissionActionFilter {
+  override implicit val executionContext: ExecutionContext = ec
+  override val testAccess: String => Authorization = acl.testUser(Permissions.LaunchCommercialFrontsAlert, "facia-tool-allow-access-for-all")
+  override val restrictedAction: String = "Launch commercial fronts."
+}
+
+class LaunchEditorialFrontsPermissionCheck(val acl: Acl)(implicit ec: ExecutionContext) extends PermissionActionFilter {
+  override implicit val executionContext: ExecutionContext = ec
+  override val testAccess: String => Authorization = acl.testUser(Permissions.LaunchEditorialFrontsAlert, "facia-tool-allow-access-for-all")
```

Comment:
Is `facia-tool-allow-access-for-all` correct?

Reply:
They are not. I will update them